src/code/z_lights.c

```diff
@@ -223,22 +216,31 @@ void func_8007A698(LightContext* lightCtx, u8 arg1, u8 arg2, u8 arg3, s16 numLig
     lightCtx->unk_0C = arg5;
 }
 
-Lights* Lights_New(LightContext* lightCtx, GraphicsContext* gfxCtx) {
-    return Lights_AllocAndInit(gfxCtx, lightCtx->ambient.r, lightCtx->ambient.g, lightCtx->ambient.b);
+/*
+ * Allocate a new Lights group and initilize the ambient color with that provided by LightContext
+ */
+Lights* LightContext_NewLights(LightContext* lightCtx, GraphicsContext* gfxCtx) {
+    return Lights_New(gfxCtx, lightCtx->ambient.r, lightCtx->ambient.g, lightCtx->ambient.b);
 }
 
-void Lights_InitList(GlobalContext* globalCtx, LightContext* lightCtx) {
+void LightContext_InitList(GlobalContext* globalCtx, LightContext* lightCtx) {
     lightCtx->listHead = NULL;
 }
 
-void Lights_DestroyList(GlobalContext* globalCtx, LightContext* lightCtx) {
+void LightContext_DestroyList(GlobalContext* globalCtx, LightContext* lightCtx) {
     while (lightCtx->listHead != NULL) {
-        Lights_Remove(globalCtx, lightCtx, lightCtx->listHead);
+        LightContext_RemoveNode(globalCtx, lightCtx, lightCtx->listHead);
         lightCtx->listHead = lightCtx->listHead->next;
     }
 }
 
-LightNode* Lights_Insert(GlobalContext* globalCtx, LightContext* lightCtx, LightInfo* info) {
+/*
+ * Insert a new light node into the list pointed to by LightContext
+ *
+ * Note: Due to the limited number of slots in a Lights group, inserting too many nodes in the
+ * list may result in older entries not being bound to a Light when calling Lights_BindAll
+ */
+LightNode* LightContext_InsertNewNode(GlobalContext* globalCtx, LightContext* lightCtx, LightInfo* info) {
```

Comment:
Imo this function's purpose is to add a new "light" in the list (ie the light described by `LightInfo`). The fact that it inserts a `LightNode` that points to this `LightInfo` is just a mean to do that, not the purpose. In the end, a `LightNode` is just the structure used to create what is widely known as a **doubly linked list** of `LightInfo`.
So I suggest naming this function `LightContext_InsertLight` or something along those lines, and updating its description accordingly.

`LightContext_RemoveNode` would also have to renamed accordingly ofc (`LightContext_RemoveLight` for instance).
And it would probably be good to update the description for `Lights_BindAll` to make it clear it binds lights and not the nodes themselves, which are just the linked list implementation.

Reply:
I felt it was more appropriate to make it clear that this is inserting a node into the light context list. Especially with the system name change, LightContext_InsertNode makes it clear to me that its inserting an item to the list that exists in light context

I do like LightContext_InsertLight more than just plain old "insert", cause "Light Context Insert" didnt really make sense at all. Ill probably go with this name change but wanted to explain my thought process

Reply:
Right, I understand why the name had to be changed. But my point is it's not inserting a light node (`LightNode`) in the list, it's inserting a `LightInfo` (which describes a light).
The node is just an intermediate structure used to link items in the list, not the item itself. You can think of it like a heap node VS the actual memory block it corresponds to.

Reply:
ah yeah that makes alot more sense when put that way.
Im a little worried about the separation between an "abstract" light (represented by info) and a real light from a LightsN group not being clear with a name like this, but at least externally it makes sense to refer to it as a plain old light. And the LightContext system name makes it clear its the light context list